src/main/java/com/flightstats/hub/dao/aws/DynamoUtils.java

```diff
@@ -91,25 +92,37 @@ void updateTable(String tableName, ProvisionedThroughput throughput) {
         }
     }
 
+    public boolean doesTableExist(String tableName) {
+        return getTableDescription(tableName, TableStatus.ACTIVE).isPresent();
+    }
+
     private TableDescription waitForTableStatus(String tableName, TableStatus status) {
         long endTime = System.currentTimeMillis() + TimeUnit.MINUTES.toMillis(tableCreationWaitMinutes);
         while (System.currentTimeMillis() < endTime) {
-            try {
-                TableDescription tableDescription = dbClient.describeTable(tableName).getTable();
-                if (status.equals(TableStatus.fromValue(tableDescription.getTableStatus()))) {
-                    logger.info("table " + tableName + " is " + status.toString());
-                    return tableDescription;
-                }
-            } catch (AmazonServiceException ase) {
-                logger.info("exception creating table " + tableName + " " + ase.getMessage());
-                throw ase;
+            Optional<TableDescription> existing = getTableDescription(tableName, status);
+            if (existing.isPresent()) {
+                return existing.get();
             }
             sleep();
         }
         logger.warn("table never went active " + tableName);
         throw new RuntimeException("Table " + tableName + " never went active");
     }
 
+    private Optional<TableDescription> getTableDescription(String tableName, TableStatus status) {
+        try {
+            TableDescription tableDescription = dbClient.describeTable(tableName).getTable();
+            if (status.equals(TableStatus.fromValue(tableDescription.getTableStatus()))) {
+                logger.info("table " + tableName + " is " + status.toString());
```

Comment:
Think I'm going to hold off on any more of these for now, since Manjula is doing some of this on her branch.  Will look at just doing an app-wide, focused refactor later if there're any left after M's work comes in.